float32 velocity
bool fire
bool save